### PR TITLE
Adding target to upgrade pip-tools in template

### PR DESCRIPTION
**Pull request by jinder1s**
**Description:**

Adding new target to upgrade template pip-tools.txt when upgrading edx-cookiecutters requirements. 

**JIRA:**

[XXX-XXXX](https://openedx.atlassian.net/browse/XXX-XXXX)

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,6 +26,15 @@ upgrade: ## update the requirements/*.txt files with the latest packages satisfy
...
 	$(PIP_COMPILE) -o requirements/dev.txt requirements/dev.in
 
+	make upgrade_template
+
+REQ_PATH = "python-template/{{cookiecutter.placeholder_repo_name}}/requirements"
+
+upgrade_template: export CUSTOM_COMPILE_COMMAND=make upgrade
+upgrade_template:
+	pip install -qr requirements/pip-tools.txt
+	$(PIP_COMPILE) --allow-unsafe --rebuild -o "$(REQ_PATH)/pip-tools.txt" "$(REQ_PATH)/pip-tools.in"
```
[the remark this answers is not in the document]
Comment (jinder1s) [reply Makefile]: I do not understand this comment, I will bring this up tomorrow during parking lot

[the remark this answers is not in the document]
Comment (jinder1s) [reply Makefile]: pip.in is another file on top of pip-tools that we are adding to requirements folder

File: python-template/{{cookiecutter.placeholder_repo_name}}/requirements/pip-tools.txt [hunk 2]
```diff
@@ -5,8 +5,8 @@
...
 #
 click==7.1.2              # via pip-tools
-pip-tools==5.1.2          # via -r requirements/pip-tools.in
-six==1.14.0               # via pip-tools
+pip-tools==5.4.0          # via -r python-template/{{cookiecutter.placeholder_repo_name}}/requirements/pip-tools.in
+six==1.15.0               # via pip-tools
 
 # The following packages are considered to be unsafe in a requirements file:
-# pip
+pip==20.3.1               # via pip-tools
```
Comment (jmbowman) [inline python-template/{{cookiecutter.placeholder_repo_name}}/requirements/pip-tools.txt:12]: You need to re-run the command after your edit for this to get commented out again.

File: python-template/{{cookiecutter.placeholder_repo_name}}/requirements/pip-tools.txt [hunk 3]
```diff
@@ -4,9 +4,10 @@
 #
 #    make upgrade
 #
-click==7.1.2              # via pip-tools
-pip-tools==5.4.0          # via -r python-template/{{cookiecutter.placeholder_repo_name}}/requirements/pip-tools.in
-six==1.15.0               # via pip-tools
+click==7.1.2
+    # via pip-tools
+pip-tools==5.5.0
```
Comment (jinder1s) [inline python-template/{{cookiecutter.placeholder_repo_name}}/requirements/pip-tools.txt:9]: @jmbowman Do you know why comments(# via pip-tools) is starting to show up on new line?

Comment (timmc-edx) [reply python-template/{{cookiecutter.placeholder_repo_name}}/requirements/pip-tools.txt:9]: That's new in pip-tools 5.5.0: https://github.com/jazzband/pip-tools/releases/tag/5.5.0

Comment (jmbowman) [reply python-template/{{cookiecutter.placeholder_repo_name}}/requirements/pip-tools.txt:9]: Yeah, that just started raising a few hackles over in the PR thread now that the release is out: https://github.com/jazzband/pip-tools/pull/1237#issuecomment-753984309 .  I'll be surprised if it gets reverted, but some more heads up and/or an opt-in approach would have been nice.

**Comment by jinder1s** [comment]
@jmbowman I reran upgrade here. Is there anything else that needs done?

**Comment by jmbowman** [comment]
Looks like there are some pylint warnings to fix, but seems ok other than that.

File: cookiecutter-django-app/{{cookiecutter.repo_name}}/manage.py [hunk 4]
```diff
@@ -12,18 +12,18 @@
...
             import django  # pylint: disable=unused-import, wrong-import-position
-        except ImportError:
+        except ImportError as import_error:
             raise ImportError(
                 "Couldn't import Django. Are you sure it's installed and "
                 "available on your PYTHONPATH environment variable? Did you "
                 "forget to activate a virtual environment?"
-            )
+            ) from import_error
         raise
```
Comment (jinder1s) [inline cookiecutter-django-app/{{cookiecutter.repo_name}}/manage.py:28]: While fixing the quality error, I found this empty raise line. Is there a reason for this? @jmbowman

File: Makefile [hunk 5]
```diff
@@ -26,6 +26,15 @@ upgrade: ## update the requirements/*.txt files with the latest packages satisfy
 	$(PIP_COMPILE) -o requirements/ci.txt requirements/ci.in
 	$(PIP_COMPILE) -o requirements/dev.txt requirements/dev.in
 
+	make upgrade_template
+
+REQ_PATH = "python-template/{{cookiecutter.placeholder_repo_name}}/requirements"
+
+upgrade_template: export CUSTOM_COMPILE_COMMAND=make upgrade
```
Comment (feanil) [inline Makefile:33]: The words `upgrade_template` don't tell me much about what this target is actually supposed to.  I don't understand how that means "upgrade template pip-tools.txt"

Can you add a comment and/or update the name to more clear about the purpose of this make target.

Comment (jinder1s) [reply Makefile:33]: Okay, added the comment. I kept the name the same cause I can't think of anything better in terms of conveying meaning and being concise.

File: Makefile [hunk 6]
```diff
@@ -26,6 +26,15 @@ upgrade: ## update the requirements/*.txt files with the latest packages satisfy
...
+upgrade_template: ## update the requirements/pip-tools.txt files with latest packages satisfying requirements
```
Comment (feanil) [inline Makefile:34]:
```suggestion
# update the requirements/pip-tools.txt files within our cookiecutter template code with 
# the latest packages satisfying requirements
upgrade_template: 
```

**Review by feanil — APPROVED** [review]
Good to squash and merge after the suggestion or something like it is added to the comment to make it clear which requirements file is being updated.

**Pull request merged**